jaxws-ri/runtime/rt/src/main/java/com/sun/xml/ws/fault/SOAP12Fault.java

```diff
@@ -99,7 +99,7 @@ class SOAP12Fault extends SOAPFaultBuilder {
 
     SOAP12Fault(CodeType code, ReasonType reason, String node, String role, Element detailObject) {
         this.code = code;
-        this.reason = reason;
+        this.reason = isCaptureExceptionMessage() ? reason : new ReasonType(SERVER_ERROR);
```

Comment:
reason is not String here. It is an instance of ReasonType class. Do we need to use it's default toString() implementation ?

Reply:
Ah, missed that. ReasonType contains a List of texts (min 0, max unbounded), so should all of them be checked and eventually updated to not contain server errors?

Reply:
As per me, since the fix demands all the errors be replaced with one generic error, so this should serve the purpose. And that too in this case, everywhere in SOAPFaultBuilder, new instance is created using new ReasonType(String faultString).